### ISSUE TITLE
Improve warning when there are $ in doc strings / .md files in Documenter 1.0

### ISSUE THREAD

**Issue by kellertuer** [JuliaDocs/Documenter.jl#2288]
I was a bit uncareful with my LaTeX writing in the doc strings and the markdown files, cf. e.g.

https://github.com/JuliaManifolds/Manopt.jl/blob/0865eb8b3a9a6f937c0599354bd3bbbc4c5e6c6d/docs/src/solvers/ChambollePock.md?plain=1#L6

While the “old” Documenter (pre 1.0) was quite fine with this (or tolerant to `$...$` LaTeX, now one gets the warning

```
┌ Warning: Unexpected Julia interpolation of type Symbol in the Markdown.
│   value = :p
└ @ Documenter.HTMLWriter ~/.julia/packages/Documenter/9kOxY/src/html/HTMLWriter.jl:2215
```

(which luckily someone hunted down for me to the above line). I think that warning could maybe improved?
It also seems that this does not always happen … so maybe also only for `$` in Doc strings but not in md files?
The reason I assume that is, that some of my Markdown files are generated by Quarto before the Documenter run – that Markdown contains (for laziness and Quarto/Pandoc preferring `$...$`) normal LaTeX, and that renders without warning.

Sorry if this is a bit vague, but I am still not 100% sure how to reproduce this.

**Comment by odow** [comment]
x-ref https://github.com/JuliaDocs/Documenter.jl/issues/872

**Comment by kellertuer** [comment]
Yes, kind of, while for a missing reference I can _both_ search for the reference mentioned _and_ know the file (which often at least helps me, I often see what is wrong with the ref already) – here I have neither.

But sure, even having a line number would be a dream.

**Comment by odow** [comment]
I think we could improve all error reporting, not just this specific one, or the reference one.

**Comment by mortenpi** [comment]
https://github.com/JuliaDocs/Documenter.jl/pull/2327

**Comment by odow** [comment]
Yeah I wasn't sure if the warning closes this. It's a more helpful but still not very helpful warning.

**Comment by mortenpi** [comment]
That's fair actually. Let's make having some source file / line information in the warning the actionable item here.
